proposals/NNNN-atf_facade_proposal.md

```diff
@@ -0,0 +1,255 @@
+
+So current approach leads to create test scripts which have multiple direct dependencies on different modules of ATF.
+In other words automated test scripts are fully dependent on ATF implementation and each change in ATF can lead to need for change of test scripts.
+
+In addition current approach requires a strong knowledge in ATF modules purposes and interactions between them for create good automated test scenario using ATF.
+So creation of automated test scripts takes a lot of time and it can be a bit complicated for Mobile application developers in the beginning.
+
+Creation of ATF Facade will allow:
+ - to create test scripts which are independent of ATF implementation
+ - to unify ATF use approach
+ - to document ATF functionality
+ - to improve test scripts readability and their structure on base of scenarious
+
+## Proposed solution
+
+Add new module Facade with next submodules in ATF:
+ - Runner - Manage of run test scenario
+ - SDL - Provide access to manage SDL
+ - HMI - Provide ability to emulate HMI and manage its expectations
+ - Mobile - Provide ability to emulate Mobile device and manage its expectations
+ - Utilities - Provide access to utilities of ATF
+ - Constants - Provide access to all constants used in ATF
+
+### Submodules description
+
+#### Runner - Manage of run test scenario
+
+ - Step - Add step to test scenario
+ - Title - Add titled section to test scenario
+ - TestSettings - Set special test settings for run of current test
+
+#### SDL - Provide access to manage SDL
+
+ - StartSDL - Start SDL
+ - StopSDL - Stop SDL
+ - CheckStatusSDL - Check current status of SDL
+
+#### HMI - Provide ability to emulate HMI and manage its expectations
+
+ - Connect - Create emulated HMI and connect it to SDL
+ - Disconnect - Disconnect HMI from SDL
+ - Send - Send raw message from HMI
+ - SendRequest - Send request message from HMI
+ - SendNotification - Send Notification message from HMI
+ - SendResponse - Send Response message from HMI
+ - ExpectResponse - Create expectation for response from SDL
+ - ExpectNotification - Create expectation for notification from SDL
+ - ExpectRequest - Create expectation for request from SDL
+
+#### Mobile - Provide ability to emulate Mobile device and manage its expectations
+
+ - Connect - Create emulated Mobile device and connect it to SDL
+ - Disconnect - Disconnect Mobile device from SDL
+ - StartSession - Start mobile session on Mobile device
+ - StartService - Start service on mobile session
+ - StopService - Stop service on mobile session
+ - Send - Send raw message from mobile session
+ - SendRPC - Send RPC message from mobile session
+ - ExpectResponse - Create expectation for response from SDL
+ - ExpectNotification - Create expectation for notification from SDL
+
+#### Utilities - Provide access to utilities of ATF
+
```

Comment:
@aderiabin not clear what exact functionality should be provided

Reply:
@LuxoftAKutsan Will not be added to base Facade. Can be added in addition.